articles/search/search-howto-complex-data-types.md

```diff
@@ -244,6 +244,71 @@ To filter on a complex collection field, you can use a **lambda expression** wit
 
 As with top-level simple fields, simple subfields of complex fields can only be included in filters if they have the **filterable** attribute set to `true` in the index definition. For more information, see the [Create Index API reference](/rest/api/searchservice/create-index).
 
+Azure search has the limitation wherein the complex object in all the collections total in a document cannot exceeed 3000
+
+User wil encounter the below error during indexing when there are complex collection exceeding 3000 limit
+
+“A collection in your document exceeds the maximum elements across all complex collections limit. The document with key '1052' has '4303' objects in collections (JSON arrays). At most '3000' objects are allowed to be in collections across the entire document. Remove objects from collections and try indexing the document again."
+
+In some usecases we might need to add more than 3000 items in a collection .In those cases we can pipe and store it as a string .There is no limitation of number of strings stored in an array in azure search
+
+For example if we need to search for all combination of countries and products and categories if we had stored it as complex object as below the combination cannot exceed 3000
+
+```json
+
+"searchScope": [
+  {
+     "countryCode": "FRA",
+     "productCode": 1234,
+     "categoryCode": "C100" 
+  },
+  {
+     "countryCode": "USA",
+     "productCode": 1235,
+     "categoryCode": "C200" 
+  }
+]
+```
+
+If we instead concatenate and store the values as delimited we can overcome the limitation
+```json
+"searchScope": [
+        "|FRA|1234|C100|",
+        "|FRA|*|*|",
+        "|*|1234|*|",
+        "|*|*|C100|",
+        "|FRA|*|C100|",
+        "|*|1234|C100|"
+]
+
+```
+The reason we stored it as above is because if we want to filter only on france or only on category c100 this combination with wildcard for rest of the properties will take care of that.
```

Comment:
Not sure what this means - can you please clarify?

Reply:
The reason we have stored the values with wildcards too instead of just storing them as below

>`|FRA|1234|C100|`

is to cater to search scenarios where the customer might want to search for all items which have country France irrespective of products and categories. Similarly, the customer might need to search if the item has product 1234 irrespective of the country or the category.



If the user wants to filter only on country let's say France. We will take the user input and construct it as a string as below

>`|FRA|*|*|`

which we can then use to filter in azure search as we search in an array of item values

Similarly, if user searches for France and the 1234 product code we will take the user input and construct it as a delimited string as below and match it against our search array

>`|FRA|1234|*|`

If we had stored only one entry 

>`|FRA|1234|C100|`

 without wild cards then if user wants to filter only on France we cannot convert the user input to match the searchScope array cause we don't know what all combination France is present in our searchScope array